crates/router/src/connector/threedsecureio/transformers.rs

```diff
@@ -0,0 +1,676 @@
+use api_models::payments::{DeviceChannel, ThreeDsCompletionIndicator};
+use common_utils::date_time;
+use error_stack::{report, IntoReport, ResultExt};
+use iso_currency::Currency;
+use isocountry;
+use masking::{ExposeInterface, PeekInterface, Secret};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    connector::utils::{to_connector_meta, AddressDetailsData, CardData, SELECTED_PAYMENT_METHOD},
+    consts::NO_ERROR_MESSAGE,
+    core::errors,
+    types::{
+        self,
+        api::{self, MessageCategory},
+        authentication::ChallengeParams,
+        transformers::ForeignTryFrom,
+    },
+    utils::OptionExt,
+};
+
+pub struct ThreedsecureioRouterData<T> {
+    pub amount: String,
+    pub router_data: T,
+}
+
+impl<T>
+    TryFrom<(
+        &types::api::CurrencyUnit,
+        types::storage::enums::Currency,
+        i64,
+        T,
+    )> for ThreedsecureioRouterData<T>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        (_currency_unit, _currency, amount, item): (
+            &types::api::CurrencyUnit,
+            types::storage::enums::Currency,
+            i64,
+            T,
+        ),
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            amount: amount.to_string(),
+            router_data: item,
+        })
+    }
+}
+
+impl<T> TryFrom<(i64, T)> for ThreedsecureioRouterData<T> {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(router_data: (i64, T)) -> Result<Self, Self::Error> {
```

Comment:
```suggestion
    fn try_from((amount, router_data): (i64, T)) -> Result<Self, Self::Error> {
```

For better readability